service/internal/graph/graph.go

```diff
@@ -196,6 +196,23 @@ func (g *Graph) createNodes(set Settings) error {
 	return nil
 }
 
+// formatPipelineNameWithSignal formats pipeline signal types with their names.
+func formatPipelineNameWithSignal(pipelineIDs []pipeline.ID, signal pipeline.Signal) string {
+	var formatted string
+
+	for _, pid := range pipelineIDs {
+		if pid.Signal().String() == signal.String() {
```

Comment:
Don't compare strings for enum like types, just compare the signal:

```suggestion
		if pid.Signal() == signal {
```